modules/post/multi/gather/tomcat_gather.rb

```diff
@@ -85,61 +93,84 @@ def gatherwin()
           end
         end
 
-        port_path = tomcat_home.split('"')[1] + "\\conf\\server.xml"
+        port_path = tomcat_home + "\\conf\\server.xml"
         if exist?(port_path)
           xml = read_file(port_path).split("\n")
         end
         comment_block = false
         xml.each do |line|
           if line.include? "<Connector" and not comment_block
-            $port = line.split('<Connector port="')[1].split('"')[0].to_i
+            i=0
+            while i < $username.count
+              $port.push(line.split('<Connector port="')[1].split('"')[0].to_i)
+              i+=1
+            end
           elsif line.include? ("<!--")
             comment_block = true
           elsif line.include? ("-->") and comment_block
             comment_block = false
           end
         end
       end
+    else
+      print_status("No Tomcat home can be determined")
     end
   end
 
   def gathernix()
     print_status("Unix OS detected")
     user_files = cmd_exec('locate tomcat-users.xml').split("\n")
-    user_files.each do |path|
-      if exist?(path) and not path.include? "tomcat8"
-          print_status("tomcat-users.xml found")
-          xml = read_file(path).split("\n")
+    if user_files.size > 0 
+      user_files.each do |path|
+        if exist?(path)
+            print_status(path + " found")
+            begin
+              xml = read_file(path).split("\n")
+            rescue
+              print_status("Cannot open " + path + " you probably don't have permission to open the file.")
```

Comment:
print_status("Cannot open #{path} you probably don't have permission to open the file.")